ckanext/datastore/controller.py

```diff
@@ -32,14 +39,24 @@ def dump(self, resource_id):
             limit = int_validator(request.GET.get('limit'), {})
         except Invalid as e:
             abort(400, u'limit: ' + e.error)
+        bom = boolean_validator(request.GET.get('bom'), {})
+        fmt = request.GET.get('format', 'csv')
 
-        wr = None
-        while True:
-            if limit is not None and limit <= 0:
-                break
+        def start_writer(columns):
+            if fmt == 'csv':
+                return csv_writer(response, columns, resource_id, bom)
+            if fmt == 'tsv':
+                return tsv_writer(response, columns, resource_id, bom)
+            if fmt == 'json':
+                return json_writer(response, columns, resource_id, bom)
+            if fmt == 'xml':
+                return xml_writer(response, columns, resource_id, bom)
+            abort(400, _(
+                u'format: must be one of %s') % u', '.join(DUMP_FORMATS))
```

Comment:
If we want to add a plugin interface for other formats (or overriding formats) this would be the place, but let's do that as after this PR to keep this change small.